lib/services/playground-service.ts

```diff
@@ -0,0 +1,50 @@
+export class PlaygroundService implements IPlaygroundService {
+	constructor(private $fs: IFileSystem,
+		private $projectDataService: IProjectDataService,
+		private $userSettingsService: IUserSettingsService) { }
+
+	public async getPlaygroundInfo(projectDir?: string): Promise<IPlaygroundInfo> {
+		const projectData = this.getProjectData(projectDir);
+		if (projectData) {
+			const projectFileContent = this.$fs.readJson(projectData.projectFilePath);
+			if (this.hasPlaygroundKey(projectFileContent)) {
+				const id = projectFileContent.nativescript.playground.id;
+				let usedTutorial = projectFileContent.nativescript.playground.usedTutorial || false;
+
+				// In case when usedTutorial=true is already saved in userSettings file, we shouldn't overwrite it
+				const playgroundInfo = await this.getPlaygroundInfoFromUserSettingsFile();
+				if (playgroundInfo && playgroundInfo.usedTutorial === true) {
+					usedTutorial = true;
+				}
+
+				delete projectFileContent.nativescript.playground;
+				this.$fs.writeJson(projectData.projectFilePath, projectFileContent);
+
+				const result = { id , usedTutorial };
+				await this.$userSettingsService.saveSettings(<any>{playground: result});
+				return result;
+			}
+
+			return await this.getPlaygroundInfoFromUserSettingsFile();
```

Comment:
I believe this line should be outside of the `if` statement as we want to return the settings from userSettingsFile even when the command is not executed in a project (i.e. when there's no projectData.

Reply:
also you do not need `await` after return.